src/backend/distributed/executor/intermediate_results.c

```diff
@@ -234,28 +236,13 @@ RemoteFileDestReceiverStartup(DestReceiver *dest, int operation,
 
 	const char *resultId = resultDest->resultId;
 
-	const char *delimiterCharacter = "\t";
-	const char *nullPrintCharacter = "\\N";
-
 	List *initialNodeList = resultDest->initialNodeList;
 	ListCell *initialNodeCell = NULL;
 	List *connectionList = NIL;
 	ListCell *connectionCell = NULL;
 
 	resultDest->tupleDescriptor = inputTupleDescriptor;
-
-	/* define how tuples will be serialised */
-	CopyOutState copyOutState = (CopyOutState) palloc0(sizeof(CopyOutStateData));
-	copyOutState->delim = (char *) delimiterCharacter;
-	copyOutState->null_print = (char *) nullPrintCharacter;
-	copyOutState->null_print_client = (char *) nullPrintCharacter;
-	copyOutState->binary = CanUseBinaryCopyFormat(inputTupleDescriptor);
```

Comment:
Would the new approach work correctly with OIDs (e.g. of custom types) not matching on different workers?

Reply:
My guess is yes, since their in-memory representation should be the same irrespective of the OID. I have to try and  look into this in more detail though.

Reply:
Arrays store the oid in their in-memory representation